MichaelAquilina/flake8-spellcheck
# General uplift of package

- Remove dependency on importlib
    - Pathlib has been in the standard library since python3.4, and so is widely available without polyfills.
- General codebase improvements
    - Add type hints to just about everything.
    - Use an enum in place of arbitrary strings for determining word case.
    - Use immutable sets for data loaded from option parsing.
    - Add an 'else: return' branch in one place to make the type checker happy.
- Update argument handling
    - Start the migration to only using namespaced flag names.
    - Start the migration away from the term "whitelist".
    - Log deprecation warnings for anyone relying on the old behaviour.
    - Take advantage of argparse functionality in the new flags, because flake8 has migrated away from optparse.
    - Update readme accordingly
- Updates to dictionary lists
	- Add 'allowlist' and 'denylist' to english dictionary list.
    - Create dedicated pytest list
    - Add some new entries to the python list:
      - 'checkable' comes from '@runtime_checkable', a decorator introduced.
        in python3.8 and also available in typing_extensions.
      - 'exc' comes from the default argument names for '__exit__'
        functions.
      - 'tokenize' is a stdlib module used by this package.
      - 'unittest' is a stdlib module.
    - Remove a couple of duplicate entries from the technical list.
    - Add some new entries to the technical list:
      - 'filesystem' should be obvious.
      - 'func' is common shorthand for 'function'.
      - 'lua' is the name of an embeddable programming language.
      - 'runtime' is a common word to differentiate from e.g. compile-time.
        It's also used in the '@runtime_checkable' decorator.
      - 'ttl' stands for Time-to-Live, and is a common acronym seen in many
        places, including DNS and caching.
      - 'util' is a commonly used name in programming for a file full of
        utility functions.
    - Removed some entries from this package's custom allowlist that are
      already specified elsewhere.

Closes #44
